### PR TITLE
Import obi dialgos

### PR DESCRIPTION
**Pull request by LarsLiden**

### CODE REVIEW

File: src/Utils/dialogUtils.tsx [hunk 1]
```diff
@@ -261,9 +261,13 @@ export function dialogSampleInput(dialog: CLM.TrainDialog | CLM.LogDialog): stri
     let round = 0
     let length = 0
     while (round < dialog.rounds.length && length < MAX_SAMPLE_INPUT_LENGTH) {
-        const userInput =
-            (dialog as CLM.LogDialog).rounds[round].extractorStep.text ||
-            (dialog as CLM.TrainDialog).rounds[round].extractorStep.textVariations[0].text
+        let userInput = (dialog as CLM.LogDialog).rounds[round].extractorStep.text
+        if (!userInput && (dialog as CLM.TrainDialog).rounds[round].extractorStep.textVariations[0]) {
+            userInput = (dialog as CLM.TrainDialog).rounds[round].extractorStep.textVariations[0].text
```
Comment (mattmazzola) [inline src/Utils/dialogUtils.tsx]: Given `(dialog as CLM.TrainDialog).rounds[round].extractorStep` is repeated 3 times consider saving to variable.

The type assertion doesn't actually check if it's a log or train dialog round only forces it.

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "axios": "^0.19.0",
     "botbuilder": "4.1.7",
     "botframework-directlinejs": "^0.9.17",
+    "clone": "^2.1.2",
```
Comment (mattmazzola) [inline package.json]: What is this used for?

File: src/Utils/obiDialogParser.ts [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                                    // Add extractor step to all the children
+                                    childDialogs.forEach(td => {
+                                        td.rounds[0].extractorStep = extractorStep
+                                    })
+    
+                                    // Add children to train dialog list
+                                    trainDialogs = [...trainDialogs, ...childDialogs]
+                                }
+                                else if (step.$type === "Microsoft.SendActivity") {
+
```
Comment (mattmazzola) [inline src/Utils/obiDialogParser.ts]: Something supposed to happen here? This is weird to be empty. Would at least put a comment saying it's intentionally skipping this `$type`, will be added later, or if expected not to happen do some sort of `throw new Error('Not Implemented')`

File: src/Utils/obiDialogParser.ts [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+                }
+            }
+        }
+        let trainRound: CLM.TrainRound | undefined
+        if (obiDialog.steps) {
+            for (const step of obiDialog.steps) {
+                if (typeof step === "string") {
+                    throw new Error("Unexected step of type string")
+                }
+                else if (step.$type === "Microsoft.SendActivity") {
```
Comment (mattmazzola) [inline src/Utils/obiDialogParser.ts]: Would move these to some `ObiStepType` enum.

File: src/routes/Apps/AppsListComponent.tsx [hunk 5]
```diff
@@ -313,6 +308,25 @@ export class Component extends React.Component<Props, ComponentState> {
             </div>
         </div>
     }
+
+    private getSortedApplications(sortColumn: ISortableRenderableColumn, apps: CLM.AppBase[]): CLM.AppBase[] {
```
Comment (mattmazzola) [inline src/routes/Apps/AppsListComponent.tsx]: Why was this moved? The whole ordering of public, private stuff seems like a weaker convention to follow than the stronger convention of having `render` be the last method in `React`. Not everyone is using TypeScript or private. Moving code is harder for code review / diffs as you can't tell was it moved and changed for just moved.

File: src/types/obiTypes.ts [hunk 6]
```diff
@@ -0,0 +1,571 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+///////////////////////////////////
+// GENERATED FILE from OBI schema
```
Comment (mattmazzola) [inline src/types/obiTypes.ts]: What the instructions to regenerate this? Would put them in the comment here.

File: package.json (hunk 2 above)
Comment (LarsLiden) [reply package.json]: Nice catch.  Had to install it for some BF stuff that I ended up removing.

File: src/routes/Apps/AppsListComponent.tsx (hunk 5 above)
Comment (LarsLiden) [reply src/routes/Apps/AppsListComponent.tsx]: If we make that choice we should add an exclusion rule is TSLINT so that it stops showing errors

File: src/Utils/obiDialogParser.ts [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+        if (obiDialog.rules) {
+            for (const rule of obiDialog.rules) {
+                if (rule.$type === "Microsoft.IntentRule") {
+    
+                    const textVariations = this.getTextVariations(rule.intent!)
+                    const extractorStep: CLM.TrainExtractorStep = {
+                        textVariations: textVariations
+                    }
+    
+                    if (rule.steps) {
```
Comment (tpmsr) [inline src/Utils/obiDialogParser.ts]: This function gets pretty deeply nested.  How about putting this at the top of the for loop as `if (!rule.steps) continue` ?

File: src/Utils/obiDialogParser.ts [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+        }
+        return trainDialogs
+    }
+    
+    private async getScorerStepfromActivity(prompt: string): Promise<CLM.TrainScorerStep> {
+    
+        const parsedActivity = prompt.substring(prompt.indexOf("[") + 1, prompt.lastIndexOf("]")).trim()
+        let response = this.composerDialog.lgMap.get(parsedActivity)
+        if (!response) {
+            // LARS temp to handle badly create dialog
```
Comment (tpmsr) [inline src/Utils/obiDialogParser.ts]: Suggest rephrasing as a concrete TODO

(eg, `// TODO : throw Error once parsing is implemented for all activity types`)

...and removing the commented-out throw below

File: src/Utils/obiDialogParser.ts [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+                    payload: action.payload,
+                    isTerminal: action.isTerminal,
+                    actionType: CLM.ActionTypes.TEXT,
+                    score: 1
+                }
+            }
+        }
+    
+        let scoreInput: CLM.ScoreInput = {
+            filledEntities: [],  //LARS handle filled entities from api calls
```
Comment (tpmsr) [inline src/Utils/obiDialogParser.ts]: Did you means to check the `//LARS` comments in ?

**Comment by tpmsr** [comment]
Mostly nits and some questions.  Looks like I was too late to catch you before the merge.

File: src/Utils/obiDialogParser.ts (hunk 9 above)
Comment (LarsLiden) [reply src/Utils/obiDialogParser.ts]: Yeah, they are a way for me to find actionable TODOs for myself.  We have a bunch of aspirational TODOs sprinkled in the code that make it hard to find things
